src/ophys_etl/modules/segmentation/qc/seed.py

```diff
@@ -0,0 +1,75 @@
+import h5py
+import numpy as np
+from pathlib import Path
+from typing import Union, Optional
+from matplotlib.axes import Axes
+from matplotlib.figure import Figure
+
+
+def add_seeds_to_axes(
+        figure: Figure,
+        axes: Axes,
+        seed_h5_group: Union[str, h5py.Group] = "seeding",
+        image_background: Optional[Union[np.ndarray, str]] = "seed_image",
+        seed_h5_path: Optional[Union[str, Path]] = None):
+    """plots seeds optionally on top of the seeding image
+
+    Parameters
+    ----------
+    axes: matplotlib.axes.Axes
+        the axes to which to add the seeds
+    seed_h5_group: Union[str, h5py.Group]
+        the hdf5 group or name of the hdf5 group
+    image_background: Optional[Union[np.ndarray, str]]
+        the image or name of the image dataset over which to plot the seeds
+    seed_h5_path: Optional[Union[str, Path]]
+        the path to the hdf5 file
+
+    """
+    if (seed_h5_path is not None) & (not isinstance(seed_h5_group, str)):
+        raise ValueError("for (seed_h5_path, seed_h5_group) provide either "
+                         "types (Path|str, str) or (NoneType, h5py.Group). "
+                         f"You provided ({type(seed_h5_path)}, "
+                         f"{type(seed_h5_group)}).")
+
+    # get the h5py.Group from file or directly
+    if isinstance(seed_h5_path, Path) | isinstance(seed_h5_path, str):
```

Comment:
I'm not a huge fan of mixing IO and functionality. Ideally the `add_seeds_to_axes` would not accept a `seed_h5_path` and would only allow `seed_h5_group` to be of type `h5py.Group`.

This would make testing much easier and would reduce the amount of non-relevant IO boilerplate (L29 - L40 & L74-75) in this function.

Reply:
I agree, and changed it.